apis/apiextensions/v1beta1/composition_types.go

```diff
@@ -152,6 +247,24 @@ type Patch struct {
 	// input to be transformed.
 	// +optional
 	Transforms []Transform `json:"transforms,omitempty"`
+
+	// Type can be set to refer to a named patch defined against the parent
+	// composition. If a Patch has a type set, then all other fields will be
+	// ignored and the referenced patch will be used instead.
+	// +optional
+	Type PatchType `json:"type,omitempty"`
+
+	// ID can be set on a patch of type ref to refer to a specific patch set by
+	// name.
+	// +optional
+	ID string `json:"id,omitempty"`
```

Comment:
```suggestion
	Name string `json:"name,omitempty"`
```

I think `name` is usually what's used in Kubernetes API as identifier if it's human-written. Also, `Patch` uses `name` field by default to identify resources in an array if I remember correctly.

Reply:
I agree that `name` is more common in this context (identifying an element in an array of objects). However I would be cautious as to whether `name` could be mistaken or confused with the `metadata.name` of the composed resource, rather than the name of the patch set.

This could be fine in this context, but I will almost certainly use `id` rather than `name` to identify elements within the `resources` array in https://github.com/crossplane/crossplane/issues/1909 (where `name` is at a higher risk of being interpreted to mean "the name of the resource" rather than "the name of the template"), and thus  prefer `id` to keep the `Composition` API internally consistent.

Reply:
Per https://github.com/crossplane/crossplane/issues/1974#issuecomment-733146966 we decided to go with `name` and `patchSetName` here. :)